daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/v2/ledgerinteraction/IdeLedgerClient.scala

```diff
@@ -317,7 +353,36 @@ class IdeLedgerClient(
             )
         }
 
-      loop(result)
+      // We use try + unsafePreprocess here to avoid the addition template lookup logic in `preprocessApiCommands`
+      val eitherSpeedyCommands =
+        try {
+          Right(preprocessor.unsafePreprocessApiCommands(commands.to(ImmArray)))
+        } catch {
+          case Error.Preprocessing.Lookup(
+                LookupError.NotFound(Reference.Package(_), Reference.Template(templateId))
+              ) =>
+            Left(makeNoSuchTemplateError(templateId))
+        }
```

Comment:
I think we should handle lookup generally. 
1- You should not assume the preprocessing will throw you a LookupError where the `context` is  `Reference.Template`. The preprocessing is free to lookup for what it want (it could be module, definition, data-type, choice)
2 - Without looking at the code, I think your code does only handle create, I except exercise will trow an error where the context is `Reference.Choice` or `Reference.TemplateOrInterface`
3 - With the current upgrade story, we will have lookup errors where the `notFound` field will not be not a `Reference.Package` 

I think instead of creating a new ad-hoc error, we would probably better catch all LookupErrors. Ideally I would  wrap the `language.LookupError` instead a new `scenario.Error`, but at first adding a new `scenario.Error` that wraps the message of the `language.LookupError` would be enough for me.

Reply:
Yeah that makes sense, I'm happy to do this, as long as we're happy to expose these errors to users. (I was unsure how internal they are)

Reply:
Right okay, I'll generalise this logic for all lookup errors then